src/binding/c/mpit_api.txt

```diff
@@ -0,0 +1,515 @@
+    attempts to start all variables within the session identified by the parameter session for
+    which handles have been allocated. In this case, the routine returns MPI_SUCCESS if all
+    variables are started successfully, otherwise MPI_T_ERR_PVAR_NO_STARTSTOP is returned.
+    Continuous variables and variables that are already started are ignored when
+    MPI_T_PVAR_ALL_HANDLES is specified.
+*/
+{
+    if (handle == MPI_T_PVAR_ALL_HANDLES) {
+        MPIR_T_pvar_handle_t *hnd;
+        DL_FOREACH(session->hlist, hnd) {
+            if (!MPIR_T_pvar_is_continuous(hnd) && !MPIR_T_pvar_is_started(hnd))
+                mpi_errno = MPIR_T_pvar_start_impl(session, hnd);
+        }
+    } else {
+        if (handle->session != session) {
+            mpi_errno = MPI_T_ERR_INVALID_HANDLE;
+            goto fn_fail;
+        }
+        if (MPIR_T_pvar_is_continuous(handle)) {
+            mpi_errno = MPI_T_ERR_PVAR_NO_STARTSTOP;
+            goto fn_fail;
+        }
+
+        if (!MPIR_T_pvar_is_started(handle)) {
+            mpi_errno = MPIR_T_pvar_start_impl(session, handle);
+            if (mpi_errno != MPI_SUCCESS)
+                goto fn_fail;
+        }
+    }
+}
+
+MPI_T_pvar_stop:
+    .desc: Stop a performance variable
+/*
+    Notes:
+    This functions stops the performance variable with the handle identified by the parameter
+    handle in the session identified by the parameter session.
+    
+    If the constant MPI_T_PVAR_ALL_HANDLES is passed in handle, the MPI implementation
+    attempts to stop all variables within the session identified by the parameter session for
+    which handles have been allocated. In this case, the routine returns MPI_SUCCESS if all
+    variables are stopped successfully, otherwise MPI_T_ERR_PVAR_NO_STARTSTOP is returned.
+    Continuous variables and variables that are already stopped are ignored when
+    MPI_T_PVAR_ALL_HANDLES is specified.
+*/
+{
+    /* If handle is MPI_T_PVAR_ALL_HANDLES, dispatch the call.
+     * Otherwise, do correctness check, then go to impl.
+     */
+    if (handle == MPI_T_PVAR_ALL_HANDLES) {
+        MPIR_T_pvar_handle_t *hnd;
+        DL_FOREACH(session->hlist, hnd) {
+            if (!MPIR_T_pvar_is_continuous(hnd) && MPIR_T_pvar_is_started(hnd)) {
+                mpi_errno = MPIR_T_pvar_stop_impl(session, hnd);
+                if (mpi_errno != MPI_SUCCESS)
+                    goto fn_fail;
+            }
+        }
+    } else {
+        if (handle->session != session) {
+            mpi_errno = MPI_T_ERR_INVALID_HANDLE;
+            goto fn_fail;
+        }
+        if (MPIR_T_pvar_is_continuous(handle)) {
+            mpi_errno = MPI_T_ERR_PVAR_NO_STARTSTOP;
+            goto fn_fail;
+        }
+
+        if (MPIR_T_pvar_is_started(handle)) {
+            mpi_errno = MPIR_T_pvar_stop_impl(session, handle);
+            if (mpi_errno != MPI_SUCCESS)
+                goto fn_fail;
+        }
+    }
+}
+
+MPI_T_pvar_write:
```

Comment:
missing
```
.error: MPI_T_ERR_PVAR_NO_WRITE
```
perhaps that block should be moved into the error checking block.

Reply:
Both are fixed.